sycl/include/CL/sycl/INTEL/esimd/detail/emu/esimdcpu_device_interface.hpp

```diff
@@ -0,0 +1,37 @@
+//==----- esimdcpu_device_interface.hpp - DPC++ Explicit SIMD API ---------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+/// \file esimdcpu_device_interface.hpp
+/// Declarations for ESIMD_CPU-device specific definitions.
+/// ESIMD intrinsic and LibCM functionalities required by intrinsic defined
+///
+/// \ingroup sycl_pi_esimd_cpu
+
```

Comment:
It seems this header should be moved somewhere into source dir since include dir contains headers that need to be compiled together with user's source files.

Reply:
This interface definition is used by intrinsic implementation as Konst put as sample above. The implementation will be added later.

```
// intrinsic implementaiton:
#ifndef __SYCL_DEVICE_ONLY__
void __esimd_barrier() {
  ESIMDDeviceInterface *I = reinterpret_cast<ESIMDDeviceInterface*>(getPluginSpecificData(esimd_emu));
  I->barrier();
}
#endif
```

Reply:
Could you please clarify what is the reason for having this file in the `include` dir?

Reply:
Is there a header file in `include` directory which should include this header? If no, please move it to another place, for example to the `source` directory.

Reply:
When ESIMD_CPU implementation is fully pushed, this header files will be used by two files
`sycl/plugins/esimd_cpu/pi_esimd_cpu.hpp`
`sycl/include/CL/sycl/INTEL/esimd/detail/esimd_memory_intrin.hpp`

While the first one uses this header file during toolchain build, the second one uses the file during kernel build as kernels will use intrinsics provided by CM.